### ISSUE TITLE
Key shared subscription: hash ranges are not fully reported in stats

### ISSUE THREAD

**Issue by eolivelli** [apache/pulsar#14073]
**Describe the bug**
When I run `pulsar-admin topics stats TOPIC-NAME` and I have a KeyShared subscription with stickyHashRanges, with many ranges, the stats are not fully reporting the ranges, it looks like that for each consumer they report only the last set of ranges assigned to each Consumer and for some Consumers the list is empty even if the Consumer set the range and it is actually consuming messages.

There should be some bug here:
https://github.com/apache/pulsar/blob/9abd6d30f39f74e255cd5dac40b77af3dce5c468/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/HashRangeExclusiveStickyKeyConsumerSelector.java#L69

  

```
 public Map<Consumer, List<Range>> getConsumerKeyHashRanges() {
        Map<Consumer, List<Range>> result = new HashMap<>();
        Map.Entry<Integer, Consumer> prev = null;
        for (Map.Entry<Integer, Consumer> entry: rangeMap.entrySet()) {
            if (prev == null) {
                prev = entry;
            } else {
                if (prev.getValue().equals(entry.getValue())) {
                    result.computeIfAbsent(entry.getValue(), key -> new ArrayList<>())
                            .add(Range.of(prev.getKey(), entry.getKey()));
                }
                prev = null;
            }
        }
        return result;
    }
```

**Additional context**
Pulsar 2.8.2

**Comment by mattisonchao** [comment]
Hi, @eolivelli 

I can help to fix it.~

**Comment by eolivelli** [comment]
@mattisonchao I know that @nicoloboschi  is already working on a fix

**Comment by mattisonchao** [comment]
@eolivelli  Ah, sorry. I just made a PR, or I'll close it...

**Comment by eolivelli** [comment]
don't need to close it
https://github.com/apache/pulsar/pull/14077

**Comment by nicoloboschi** [comment]
After having a more detailed look with @eolivelli, we've seen the version used by Enrico is a custom fork based on Pulsar 2.8.0 and it doesn't include https://github.com/apache/pulsar/pull/12212, which is the resolution to the problem

@mattisonchao you can close the PR, the problem has been already fixed. Thanks

**Comment by codelipenghui** [comment]
The issue had no activity for 30 days, mark with Stale label.

**Comment by eolivelli** [comment]
fixed by #12212
